src/coreclr/interpreter/compiler.cpp

```diff
@@ -2113,6 +2117,10 @@ int InterpCompiler::GenerateCode(CORINFO_METHOD_INFO* methodInfo)
 
     codeEnd = m_ip + m_ILCodeSize;
 
+    // Safepoint at each method entry. This could be done as part of a call, rather than
+    // adding an opcode.
+    AddIns(INTOP_SAFEPOINT);
```

Comment:
Is there any reason to not to do it as part of the call that made you add the instruction instead?

Reply:
Not a very serious reason. Felt cleaner to have a separate instruction for now, given calls in general are an area that will keep suffering modifications in the future.

Reply:
Regular CoreCLR JIT expects the safe point to be at the callsite, not in the method entry. 

This mismatch between the regular CoreCLR JIT suspension model and interpreter suspension model can lead to GC starvation (GC suspension taking a long time in corner cases). It may be better to inserts the safe point after the call to avoid these subtle issues.